torch_geometric/loader/neighbor_loader.py

```diff
@@ -25,19 +25,24 @@ def __init__(
         directed: bool = True,
         input_type: Optional[Any] = None,
         share_memory: bool = False,
+        time_attr: str = None,
     ):
         self.data_cls = data.__class__
         self.num_neighbors = num_neighbors
         self.replace = replace
         self.directed = directed
+        self.node_time = None
 
         if isinstance(data, Data):
             # Convert the graph data into a suitable format for sampling.
             out = to_csc(data, device='cpu', share_memory=share_memory)
             self.colptr, self.row, self.perm = out
             assert isinstance(num_neighbors, (list, tuple))
+            # TODO: time_attr for homogeneous graphs
 
         elif isinstance(data, HeteroData):
+            if time_attr is not None:
+                self.node_time = data[time_attr]
```

Comment:
This should be a dict, right? I think we need to do `self.node_time_dict = data.collect(time_attr)` instead.